### DISCUSSION TITLE
Add an option to flux-cli to use the namespace declared in the kubeconfig context

### DISCUSSION THREAD

**Discussion by pmialon** [fluxcd/flux2#2768] [Proposals] [answered]
Hi,

It will be great to add an option to flux-cli to use the namespace declared in the [kubeconfig context](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#setting-the-namespace-preference)

This will allow us, when we work on specific namespace, to use both kubectl and flux without specifying -n.

**Comment by stefanprodan** [comment, marked as answer]
Right now we allow changing the default namespace from `flux-system` to something else with:

```bash
export FLUX_SYSTEM_NAMESPACE=my-namespace
```

We could introduce a new env var that would make the CLI read the namespace from the kubeconfig current context e.g.:

```bash
export FLUX_SYSTEM_NAMESPACE_FROM_CONTEXT=true
```

**Reply by hiddeco** [reply]
I think a new knob to make the CLI take the context into account would not really improve the situation.

Putting the difficulties around upgrade paths and backwards compatibility aside for a minute. I think the best option in terms of UX would be to default to `flux-system` if the KubeConfig context namespace is empty or equals `default`, and otherwise use the defined namespace as the current selected namespace.

This would make tools like `kubens` work seamlessly for both `kubectl` and the Flux CLI, while maintaining domain specific defaults. Installation into the `default` namespace could e.g. still be possible by explicitly defining `-n default`.

**Reply by jybp** [reply]
Suggestion implemented in https://github.com/fluxcd/flux2/pull/5297
I like the env var idea as it doesn't introduce any breaking changes.
